util/plotFixedOscLLHWhole.py

```diff
@@ -0,0 +1,202 @@
+
+    # Histogram 2D
+    hist2d, xedges, yedges = np.histogram2d(
+        theta, deltam, bins=[theta_bins, deltam_bins], weights=llh_2delta
+    )
+    counts, _, _ = np.histogram2d(theta, deltam, bins=[theta_bins, deltam_bins])
+    # element wise: gist2d/counts, and avoiding zero in counts(If counts = 0, output = 0)
+    hist2d_avg = np.divide(hist2d, counts, out=np.zeros_like(hist2d), where=counts != 0)
+
+    # Profiles
+    profile_theta = np.min(hist2d_avg, axis=1)  # Min over deltam
+    profile_deltam = np.min(hist2d_avg, axis=0)   # Min over theta
+    theta_centers = 0.5 * (xedges[:-1] + xedges[1:])
+    deltam_centers = 0.5 * (yedges[:-1] + yedges[1:])
+    #print(profile_theta-profile_deltam)
+
+    # Plot setup
+    SNOPLUSSTYLE()
+    fig = plt.figure(figsize=(10, 8))
+    gs = gridspec.GridSpec(2, 2, width_ratios=[3, 1], height_ratios=[1, 3], hspace=0.0, wspace=0.0)
+
+    # Share x and y axes as needed
+    ax_heat = fig.add_subplot(gs[1, 0]) #2D LLH
+    ax_top = fig.add_subplot(gs[0, 0]) # theta12 
+    ax_right = fig.add_subplot(gs[1, 1]) # delm12
+    cbar_ax = fig.add_subplot(gs[0, 1])  #colorbar
+
+    # Setting Minor Tick On
+    ax_heat.minorticks_on()
+    ax_top.minorticks_on()
+    ax_right.minorticks_on()
+    cbar_ax.minorticks_on()
+    # Top left: θ12 profile
+    ax_top.plot(theta_centers, profile_theta, color='black')
+    ax_top.set_xlim(theta_centers.min(), theta_centers.max())
+    ax_right.set_label(r"$2\,\Delta\ln\mathcal{L}$")
+    ax_top.set_xticklabels([])
+
+    # Bottom left: 2D LLH plot
+    X, Y = np.meshgrid(xedges, yedges)
+    ax_heat.yaxis.set_major_formatter(mticker.FuncFormatter(scaled_formatter))
+    ax_heat.set_ylabel(r"$\Delta m_{21}^2$ ($\times 10^{-5}$ MeV$^2$)")
+    pc = ax_heat.pcolormesh(X, Y, hist2d_avg.T, shading='auto', cmap='viridis')
+    ax_heat.set_xlabel(r"$\theta_{12}$")
+    # plot contour for 1 and 2 sigma
+    # Contour levels: 1σ and 2σ
+    contour_levels = [2.30, 6.18] 
+    CS = ax_heat.contour(
+        theta_centers, deltam_centers, hist2d_avg.T,  # Note: need centers, not bin edges
+        levels=contour_levels,
+        colors=['white', 'white'],
+        linewidths=2,
+        linestyles=['solid', 'dashed']
+    )
+    fmt = {2.30: r"$1\sigma$", 6.18: r"$2\sigma$"}
+    ax_heat.clabel(CS, fmt=fmt, inline=True, colors='black')
+
+    # Bottom right: deltam profile
+    ax_right.plot(profile_deltam, deltam_centers, color='black')
+    ax_right.set_ylim(deltam_centers.min(), deltam_centers.max())
+    ax_right.set_label(r"$2\,\Delta\ln\mathcal{L}$")
+    ax_right.set_yticklabels([])
+
+    # Top right: colorbar
+    fig.colorbar(pc, cax=cbar_ax, orientation='vertical')
+    cbar_ax.yaxis.set_ticks_position('right')
+    cbar_ax.set_label(r"$2\,\Delta\ln\mathcal{L}$")
+
+    #plt.tight_layout()
+    #plt.show()
+    plt.savefig(os.path.join(output_dir, "OscLLH.png"))
+    plt.savefig(os.path.join(output_dir, "OscLLH.pdf"))
```

Comment:
maybe call it like FullLLH or LLHandProjections, or something just to make it clear from the name what makes it different to the outputs of plotFixedOscLLH.C